### DISCUSSION TITLE
Adding additional control into itemspanel

### DISCUSSION THREAD

**Discussion by mohdimas** [AvaloniaUI/Avalonia#13693] [Q&A]
hi guys,

I have a specific use case for creating a tokenized textbox similar to [this](https://github.com/puppetsw/AvaloniaTokenizingTextBox). There's small difference, which our version has the textbox in the same wrap panel as the items. On the `0.10.x` this is our template:
```
<ItemsPresenter Name="PART_ItemsPresenter"
                                        Items="{TemplateBinding Items}">
                            <ItemsPresenter.ItemsPanel>
                                <ItemsPanelTemplate>
                                    <WrapPanel>
                                        <TextBox />
                                    </WrapPanel>
                                </ItemsPanelTemplate>
                            </ItemsPresenter.ItemsPanel>
                        </ItemsPresenter>
```

It worked on there where we can have the `TextBox` as the last item in the same `WrapPanel` as the items. But it no longer works on `11.0.x`. So I'm just looking for inspiration, maybe you guys have ideal way to achieve that?

Thanks!

**Comment by mohdimas** [comment]
The workaround is by adding the item to the `LogicalChildren`. Added a sample [here](https://github.com/mohdimas/AvaloniaTest/tree/logical-children-issue). The first `ItemsControl` doesn't render the `TextBox` and the second one is rendered correctly.

<img width="1462" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/bdf6d7d0-0007-4be1-98e1-4aa62758a794">

In the dev tools, the first textbox is also added but contains no style or template 

<img width="1392" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/e1122e3b-e37c-4f0f-a1ed-aab7d4f53d0a">

**Comment by timunie** [comment]
Imo better to add a "blank" item to the Collection bound to and use a special DataTemplate for it.

**Reply by mohdimas** [reply]
yes there are multiple ways to achieve that. but I'm just pointing out that this no longer works in `11.0.x` where it worked before in previous version.

**Reply by timunie** [reply]
Yeah in 11.0 we had several breaking changes. And this is one of those. It was never really an intended feature as it will not work with virtualized lists anyway, and most are virtualized by default.
